traj/src/traj/seven_segment.py

```diff
@@ -0,0 +1,90 @@
+from sympy import integrate, Symbol
+from sympy.core.numbers import Float
+
+from .piecewise_function import PiecewiseFunction
+
+def fit_seven_segment(p_start, p_end, v_max, a_max, j_max):
+    """
+    Find the optimal seven segment trajectory for zero start and end velocities, and the given
+    start and end positions.
+
+    Follows the nomenclature and approach of
+
+        Herrera-Aguilar, Ignacio, and Daniel Sidobre. "Soft motion trajectory planning and
+        control for service manipulator robot." Workshop on Physical Human-Robot Interaction in
+        Anthropic Domains at IROS. 2006.
+    """
+    assert(a_max > 0.0)
+    assert(j_max > 0.0)
+    assert(v_max > 0.0)
+
+    # Maximum amount of time we can spend at any of our limit conditions before we violate the
+    # next higher limit condition.
+    T_jmax = a_max / j_max
+    T_amax = v_max / a_max - a_max / j_max
+
+    if T_amax < 0.0:
+        # Using max positive jerk and then max negative jerk, we don't have time to
+        # reach max acceleration before reaching max velocity. To account for this,
+        # we adjust the max acceleration down to what we can actually reach on our way
+        # to the max velocity, and we adjust the max time spent in the max jerk limited
+        # section to just reach the new max acceleration.
+        T_amax = 0.0
+        T_jmax = (v_max / j_max)**0.5
+        print(T_jmax, j_max)
+        a_max = T_jmax * j_max
+
+    # Compute the minimum distance that each case can travel. D_thr1 is the minimum distance for a
+    # trajectory that hits both max acceleration and max velocity. D_thr2 is the minimum distance
+    # for a trajectory that hits max acceleration but not max velocity.
+    D_thr1 = (a_max * v_max) / j_max + v_max**2 / a_max
+    D_thr2 = 2.0 * a_max**3 / j_max**2
+
+    D = p_end - p_start
+    if D >= D_thr1:
```

Comment:
What if abs(D) >= D_thr1, but p_start > p_end?  We'd hit both -v_max and -a_max.

Reply:
Hmmm.... I'll have to think about this. I think I'm going to go ahead and merge, knowing that this function doesn't handle all cases. Lets add a fix and a unit test for this case in a future PR.